File: csharp/DVSClient/DVSClient/address/layout/LayoutConfiguration.cs
using DVSClient.Common;

namespace DVSClient.Address.Layout
{
    /// <summary>
    /// Configuration class for setting up the address layout client.
    /// </summary>
    public class LayoutConfiguration : Configuration
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="LayoutConfiguration"/> class.
        /// </summary>
        /// <param name="builder">The builder object containing configuration settings.</param>
        protected LayoutConfiguration(LayoutBuilder builder) : base(builder)
        {
        }

        /// <summary>
        /// Creates a new builder instance for configuring the address layout client.
        /// </summary>
        /// <param name="token">The API token to authenticate requests.</param>
        /// <returns>A new instance of the <see cref="LayoutBuilder"/> class.</returns>
        public static LayoutBuilder NewBuilder(string token)
        {
            return new LayoutBuilder(token);
        }

        /// <summary>
        /// Builder class for constructing a <see cref="LayoutConfiguration"/> object with custom settings.
        /// </summary>
        public class LayoutBuilder : Builder
        {
            /// <summary>
            /// Initializes a new instance of the <see cref="LayoutBuilder"/> class.
            /// </summary>
            /// <param name="token">The API token to authenticate requests.</param>
            public LayoutBuilder(string token) : base(token)
            {
            }

            /// <summary>
            /// Builds the <see cref="LayoutConfiguration"/> object with the specified settings.
            /// </summary>
            /// <returns>A <see cref="LayoutConfiguration"/> object with the configured settings.</returns>
            public override LayoutConfiguration Build()
            {
                return new LayoutConfiguration(this);
            }

            /// <summary>
            /// Sets a custom reference ID for API requests.
            /// </summary>
            /// <param name="referenceId">The reference ID for tracking the request.</param>
            /// <returns>The current <see cref="LayoutBuilder"/> instance for method chaining.</returns>
            /// <remarks>
            /// Use this method to set a unique reference ID for tracking API requests.
            /// </remarks>
            [Obsolete("Set your reference ID as part every API interaction like the creating or deleting a layout call instead.")]
            public new LayoutBuilder SetTransactionId(string referenceId)
            {
                base.SetTransactionId(referenceId);
                return this;
            }
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/address/layout/LayoutConfiguration.java
package com.experian.dvs.client.address.layout;

import com.experian.dvs.client.common.Configuration;

/**
 * AddressConfiguration class for setting up the address layout client.
 * Provides options for customizing address layout-related API requests.
 */
public class LayoutConfiguration extends Configuration {

    /**
     * Initializes a new instance of the {@link LayoutConfiguration} class using the specified builder.
     *
     * @param builder The builder containing the configuration settings.
     */
    protected LayoutConfiguration(LayoutBuilder builder) {
        super(builder);
    }

    /**
     * Creates a new instance of the {@link LayoutBuilder} class to configure the address layout client.
     *
     * @param token The authentication token for the API.
     * @return A new instance of the {@link LayoutBuilder} class.
     */
    public static LayoutBuilder newBuilder(final String token) {
        return new LayoutBuilder(token);
    }

    /**
     * AddressBuilder class for constructing an {@link LayoutConfiguration} object with custom settings.
     */
    public static class LayoutBuilder extends Configuration.Builder {

        /**
         * Initializes a new instance of the {@link LayoutBuilder} class with the specified token.
         *
         * @param token The authentication token for the API.
         */
        public LayoutBuilder(String token) {
            super(token);
        }

        /**
         * Builds the {@link LayoutConfiguration} object with the specified settings.
         *
         * @return A new instance of the {@link LayoutConfiguration} class.
         */
        public LayoutConfiguration build() {
            return new LayoutConfiguration(this);
        }

        /**
         * Sets a custom transaction ID for API requests.
         *
         * @param referenceId The transaction ID to use.
         * @return The current {@link LayoutBuilder} instance for method chaining.
         * @deprecated Set your reference ID as part every API interaction like the creating or deleting a layout call instead.
         */
        @Deprecated(since="1.1.10", forRemoval=true)
        public LayoutBuilder setTransactionId(String referenceId) {
            super.setTransactionId(referenceId);
            return this;
        }
    }
}
